src/components/VAudioThumbnail/VAudioThumbnail.vue

```diff
@@ -49,14 +49,45 @@ export default {
       required: true,
     },
   },
-  setup() {
+  setup(props) {
     /* Switching */
 
     const ok = ref(true)
     const handleError = () => {
       ok.value = false
     }
 
+    /* PRNG */
+
+    /**
+     * Small Fast Counter is a seedable pseudo-random number generator.
+     * @see {@link https://github.com/bryc/code/blob/master/jshash/PRNGs.md#sfc32}
+     */
+    const sfc32 = (a, b, c, d) => () => {
+      a |= 0
+      b |= 0
+      c |= 0
+      d |= 0
+      let t = (((a + b) | 0) + d) | 0
+      d = (d + 1) | 0
+      a = b ^ (b >>> 9)
+      b = (c + (c << 3)) | 0
+      c = (c << 21) | (c >>> 11)
+      c = (c + t) | 0
+      return (t >>> 0) / 4294967296
+    }
+    const rand = (seed) => sfc32(0x9e3779b9, 0x243f6a88, 0xb7e15162, seed)
+    const hash = (str) => {
+      let hash = 0
+      if (str.length === 0) return hash
+      for (let i = 0; i < str.length; i++) {
+        let chr = str.charCodeAt(i)
+        hash = (hash << 5) - hash + chr
+        hash |= 0 // Convert to 32bit integer
+      }
+      return hash
+    }
```

Comment:
Would it make sense to move these into library functions instead of redefining them for every instance of the audio thumbnail?

It also occurred to me (can be a separate issue) but we can avoid calculating or spending any time on the generated artwork when the audio has a thumbnail using an early return.

Reply:
+1 to Sara's suggestion; I am using a prng in #618 and would be content to use this one instead!

Reply:
Makes sense, exporting functions into utils in 418dbfc.